### PR TITLE
Create /tools/dart-fix

### PR DESCRIPTION
**Pull request by kwalrath**
Once #2804 is merged, we can update its table to point to this page.

Staged: https://kw-staging-dartlang-2.firebaseapp.com/tools/dart-fix

Fixes #2978

### CODE REVIEW

File: src/tools/dart-fix.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+title: The dart fix command
+description: Help for bringing your code up-to-date.
+toc: false
+---
+
+The `dart fix` command finds and fixes two types of issues:
+
+* Analysis issues that have associated automated fixes
```
Comment (devoncarew) [inline src/tools/dart-fix.md:9]: We also refer to the automated fixes as 'quick fixes' - I don't know if we want to mention that name here or not.

**Review by devoncarew — COMMENTED** [review]
lgtm!

cc @bwilkerson

**Review by devoncarew — APPROVED** [review]

File: src/tools/dart-fix.md (hunk 1 above)
Comment (kwalrath) [reply src/tools/dart-fix.md:9]: Yes, I think it's good to mention synonyms that people might look for, for searchability/understability reasons. Thanks!

**Comment by devoncarew** [comment]
Actually, thinking about it, I suspect that we'll want the content of https://github.com/dart-lang/sdk/blob/master/pkg/dartdev/doc/dart-fix.md in-lined into here (and then we'd likely remove the markdown file in the sdk repo).

**Comment by kwalrath** [comment]
> Actually, thinking about it, I suspect that we'll want the content of https://github.com/dart-lang/sdk/blob/master/pkg/dartdev/doc/dart-fix.md in-lined into here (and then we'd likely remove the markdown file in the sdk repo).

If I don't get to that in this PR, I'll file a bug to do that.

A couple of questions:
* What happens if you don't include a command-line flag when you invoke `dart fix`? 
* If you don't want all of the changes it suggests, do you need to edit your code by hand? (either to undo some of the changes, or to apply/tweak the changes you want)

**Comment by devoncarew** [comment]
> * What happens if you don't include a command-line flag when you invoke `dart fix`?

It prints the help for the command and exits without making any changes on disk.

> * If you don't want all of the changes it suggests, do you need to edit your code by hand? (either to undo some of the changes, or to apply/tweak the changes you want)

It's an all-or-nothing tool. You can preview the changes that it would make with `--dry-run`. If there are some you don't want, currently you options are to apply the ones you do want by hand, or, use git, apply all the changes, and back out the ones you don't want.

File: src/tools/dart-fix.md (hunk 1 above)
Comment (bwilkerson) [reply src/tools/dart-fix.md:9]: The term 'quick-fix' was is used by IntelliJ and Eclipse. VS Code (and hence LSP) uses the term 'code action'. We should probably include both to cover a wider range of users.

**Review by bwilkerson — APPROVED** [review]

**Comment by kwalrath** [comment]
I'm going to commit this PR and file an issue to make it replace https://github.com/dart-lang/sdk/blob/master/pkg/dartdev/doc/dart-fix.md.

**Pull request merged**